apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/doedshendelse/kontrollpunkt/DoedshendelseKontrollpunktService.kt

```diff
@@ -0,0 +1,85 @@
+package no.nav.etterlatte.grunnlagsendring.doedshendelse.kontrollpunkt
+
+import kotlinx.coroutines.runBlocking
+import no.nav.etterlatte.common.klienter.PdlTjenesterKlient
+import no.nav.etterlatte.common.klienter.PesysKlient
+import no.nav.etterlatte.common.klienter.SakSammendragResponse.Companion.ALDER_SAKTYPE
+import no.nav.etterlatte.common.klienter.SakSammendragResponse.Companion.UFORE_SAKTYPE
+import no.nav.etterlatte.common.klienter.SakSammendragResponse.Status.LOPENDE
+import no.nav.etterlatte.common.klienter.SakSammendragResponse.Status.OPPRETTET
+import no.nav.etterlatte.common.klienter.SakSammendragResponse.Status.TIL_BEHANDLING
+import no.nav.etterlatte.grunnlagsendring.doedshendelse.Doedshendelse
+import no.nav.etterlatte.libs.common.behandling.SakType
+import no.nav.etterlatte.libs.common.pdl.PersonDTO
+import no.nav.etterlatte.libs.common.person.PersonRolle
+
+class DoedshendelseKontrollpunktService(
+    private val pdlTjenesterKlient: PdlTjenesterKlient,
+    private val pesysKlient: PesysKlient,
+) {
+    fun identifiserKontrollerpunkter(hendelse: Doedshendelse): List<DoedshendelseKontrollpunkt> {
+        val kontrollpunkter = mutableListOf<DoedshendelseKontrollpunkt>()
+        val avdoed = pdlTjenesterKlient.hentPdlModell(hendelse.avdoedFnr, PersonRolle.AVDOED, SakType.BARNEPENSJON)
+
+        kontrollpunkter.addNotNull(kontrollerAvdoedDoedsdato(avdoed))
+        kontrollpunkter.addNotNull(kontrollerKryssendeYtelse(hendelse))
+        kontrollpunkter.addNotNull(kontrollerDNummer(avdoed))
+        kontrollpunkter.addNotNull(kontrollerUtvandring(avdoed))
+
+        return kontrollpunkter
+    }
+
+    private fun kontrollerAvdoedDoedsdato(avdoed: PersonDTO): DoedshendelseKontrollpunkt? =
+        when (avdoed.doedsdato) {
+            null -> DoedshendelseKontrollpunkt.AvdoedLeverIPDL
+            else -> null
+        }
+
+    private fun kontrollerKryssendeYtelse(hendelse: Doedshendelse): DoedshendelseKontrollpunkt? =
+        runBlocking {
+            val kryssendeYtelser =
+                pesysKlient.hentSaker(hendelse.beroertFnr)
```

Comment:
Burde vi bare samordnet her eller bare gått mot løpende vedtak i pesys?

Reply:
Denne er ment å benyttes for å hente ut løpende vedtak i Pesys? Hva tenkte du på?